byway/src/main/webapp/routepage.js

```diff
@@ -69,60 +100,115 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
       window.alert("Could not calculate route due to: " + status);
     }
   });
+  // add starting point back into schedule panel for better visuals
+
+  route.splice(0,0,start);
+  console.log(route);
+  renderRouteList();
 }
 
-/** Clear the stops panel in the html */
-function clearStops(){
-  const stopList = document.getElementById('stop-list');
-  if(stopList != null){
-    stopList.innerText = "";
+/**
+ * Calculates and sums up the distance and time duration between all destinations (legs)
+ * @param {response} result response from the directions service object
+ */
+function computeDistanceTime(result) {
+  let totalDist = 0;
+  let totalTime = 0;
+  // full route
+  let route_response = result.routes[0];
+  let waypoint_order = route_response.waypoint_order;
+  
+  let route_copy = [...route];
+  for(let i = 0; i < route.length; i++){
```

Comment:
This snippet of code seems to be unrelated to the logic for computing the distance and time. It also doesn't look like the updated route data is used by the method. Please move this to its own section and document its purpose.

Reply:
Will do, I just did it out of convenience since the directions response was already passed in as a param, but it is bad practice.